tests/unit/cluster/extensions-parsing.tcl

```diff
@@ -0,0 +1,55 @@
+
+proc is_hostname_propagated {start_node end_node match_string} {
+    set cnt 0
+    for {set j $start_node} {$j < $end_node} {incr j} {
+        set cfg [R $j cluster slots]
+        foreach node $cfg {
+            for {set i 2} {$i < [llength $node]} {incr i} {
+                if {[string match $match_string [lindex [lindex [lindex $node $i] 3] 1]] } {
+                    incr cnt
+                }
+            }
+        }
+    }
+    return $cnt
+}
+
+start_cluster 2 2 {tags {external:skip cluster}} {
+
+    test "Cluster state should be healthy" {
+        wait_for_cluster_state "ok"
+    }
+
+    test "Set cluster hostnames on primary of shard 0 and verify they are propagated after enabling extensions processing" {
+        R 0 config set cluster-announce-hostname "host-0.com"
+
+        # Verify all nodes have received the hostname for host 0 via cluster message extensions.
+        wait_for_condition 50 100 {
+            [is_hostname_propagated 0 4 "host-0.com"] eq [llength $::servers]
+        } else {
+            fail "cluster hostname for primary 0 were not propagated to all nodes"
+        }
+
+        R 1 debug send-clustermsg-extensions 0
+        R 1 config set cluster-announce-hostname "host-1.com"
+
+        # Verify only self (node) has received the hostname.
+        wait_for_condition 50 100 {
+            [is_hostname_propagated 0 4 "host-1.com"] eq 1
+        } else {
+            fail "cluster hostname for primary 1 was propagated to other nodes"
+        }
```

Comment:
This test is not 100% correct. What we want is a permanent state where only `node-1` knows about `host-1.com`. What this test validates is that, at one point, there is one node, not necessarily  `node-1`, knows about `host-1.com`. Even without the test hook setup on L33-34, this check could pass, depending on timing. so there is a false negative bug in this logic.